### PR TITLE
Rewrote custom schema guide

### PR DESCRIPTION
**Pull request by Vultraz**
"Custom mutations" was slightly misleading since you can also add types and queries, and eventually subscriptions once they get implemented.

### CODE REVIEW

**Comment by JedWatson** [comment]
I'm strongly in favour of this, but reviewing [the preview](https://deploy-preview-2643--v5keystonejs.netlify.com/guides/custom-schema) I've got two thoughts -

### 1. Redirects for docs pages we move?

@jossmac @mitchellhamilton @timleslie has anybody thought about handling redirects? this changes the urls on the website, which could break links to the page. there are a few ways of handling this - we could add suggested pages to the 404 template, or we could keep a list of renamed pages and redirect. Haven't looked yet but I would expect gatsby has a solution for the second, the first could just be data we keep in the website for when we do stuff like this.

tbh I wouldn't necessarily block merging this PR without solving the redirect problem because there are no internal links to the page in other parts of the site and indexers will catch up -- I'm not sure it's a big problem at this point, but definitely something we should be thinking about in the future and maybe worth just sorting out now

### 2. Fix the 

@Vultraz are you open to adding better content to the page itself to cover what other schema extensions look like? what's there just covers mutations, ideally we'd have a simple example on how to add custom queries and types as well to extend the schema.

It links out to the keystone class documentation for the [extendGraphQLSchema](https://deploy-preview-2643--v5keystonejs.netlify.com/keystonejs/keystone/#extendgraphqlschemaconfig) API which is good, except the example there is pretty bad too - has the same example for the query and the mutation, and neither reference the custom type that's defined, which makes it hard to figure out how to use the features unless you're familiar enough implementing custom graphql schemas to ignore the example and just follow the API.

Probably the best case would be to have guides and examples that make sense for defining custom queries and mutations (including extending with custom types) on the "custom schema" guide page, simplify the API example on the Keystone class page and link from there to the custom schema page on how to use the API

**Comment by Vultraz** [comment]
@JedWatson I've completely rewritten the entire guide to incorporate types, queries, and mutations. Let me know if it looks good. 😃

File: docs/guides/custom-schema.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+```graphql
+mutation incrementPageViews($id: ID!) {
+  incrementPageViews(id: $id) {
+    currentViews
+    timestamp
+  }
+}
+```
+
+### A note on access control
```
Comment (Vultraz) [inline docs/guides/custom-schema.md:160]: Honestly, I think this section should probably be dropped. It's kinda tangential. Thoughts?

File: docs/guides/schema.md [hunk 2]
```diff
@@ -78,8 +78,8 @@ type User {
 }
 ```
 
-> **Note:** Only a subset of all the generated types/mutations/queries are shown
-> here. To see a more complete example [follow the Quick Start](/docs/quick-start/README.md)
+> **Note:** Only a subset of the generated types/mutations/queries are shown here.
+> For more details, see the [GraphQL introduction](https://www.keystonejs.com/guides/intro-to-graphql) page.
```
Comment (jossmac) [inline docs/guides/schema.md:82]: For internal links please use paths, which are resolved by Gatsby for the docs site, and still work when browsing the monorepo on GitHub e.g.

```suggestion
> For more details, see the [GraphQL introduction](/docs/guides/intro-to-graphql.md) page.
```

File: packages/keystone/README.md [hunk 3]
```diff
@@ -267,22 +267,24 @@ Extends keystones generated schema with custom types, queries, and mutations.
...
-      schema: 'double(x: Int): Int',
-      resolver: (_, { x }) => 2 * x,
+      schema: 'triple(x: Int): Int',
+      resolver: (_, { x }) => 3 * x,
     },
   ],
 });
 ```
 
+See the [Custom schema guide](https://www.keystonejs.com/guides/custom-schema) for more information on utilizing custom schema.
```
Comment (jossmac) [inline packages/keystone/README.md:286]: Same as [comment above](https://github.com/keystonejs/keystone/pull/2643/files#r403648772)

File: docs/guides/custom-schema.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+section: guides
+title: Custom schema
+subSection: advanced
+[meta]-->
+
+# Custom schema
+
+Out of the box, Keystone provides predictable CRUD (Create, Read, Update, Delete) GraphQL operations for [Lists](/docs/guides/schema.md#lists). They are the primary methods for updating list data and should be enough for most applications. However, custom types, queries and mutations may be added if you wish to preform non-CRUD operations.
+
+> **Hint:** See the [GraphQL philosophy](/docs/guides/graphql-philosophy.md) page for more information on how Keystone implements CRUD operations in GraphQL and when custom schema may be required.
```
Comment (jossmac) [inline docs/guides/custom-schema.md:11]: When #2654 lands, only prefixes of `'tip' | 'note' | 'important'` will get custom styles. I think I made a mistake with "hint", it feels too ambiguous and elusive.

```suggestion
> **Tip:** See the [GraphQL philosophy](/docs/guides/graphql-philosophy.md) page for more information on how Keystone implements CRUD operations in GraphQL and when custom schema may be required.
```

File: docs/guides/custom-schema.md [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+2. A custom [**mutation**](#custom-mutation) which will increment the page views.
+3. A custom [**query**](#custom-query) which will check if the current page views are over a certain threshold.
+
+> **Note:** This custom query is somewhat superfluous; you could just query the page directly and check the views client-side. However, it serves to illustrate how custom queries are set up.
+
+### Custom type
+
+First, let's define what we want our custom GraphQL type to look like:
+
+> **Hint:** You can [learn more about schemas and types](https://graphql.org/learn/schema/) on <https://graphql.org>. You can also find a reference of all the types generated by Keystone in built-in GraphQL Playground's Docs tab, accessible via the Admin UI.
```
Comment (jossmac) [inline docs/guides/custom-schema.md:74]: Same as [comment above](https://github.com/keystonejs/keystone/pull/2643/files#r403649764):

```suggestion
> **Tip:** You can [learn more about schemas and types](https://graphql.org/learn/schema/) on <https://graphql.org>. You can also find a reference of all the types generated by Keystone in built-in GraphQL Playground's Docs tab, accessible via the Admin UI.
```

**Comment by Vultraz** [comment]
@jossmac done 👍

**Comment by Vultraz** [comment]
Resolved conflicts.

File: docs/guides/custom-mutations.md [hunk 6]
```diff
@@ -1,143 +0,0 @@
-<!--[meta]
-section: guides
-title: Custom mutations
-subSection: advanced
-[meta]-->
-
```
Comment (jesstelford) [inline docs/guides/custom-mutations.md:6]: Suggestion: Keep this page, but remove all of its contents and replace it with:

```
This page has been moved to [/guides/custom-schema](./custom-schema).
```

**Comment by jesstelford** [comment]
I've created an issue to track the redirects which I think is a good idea to do: https://github.com/keystonejs/keystone/issues/2671

**Pull request merged**